File: Map/Map/Map/domain/OutList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Map.domain
{
    public class OutList<T> : ILis<T>
    {
        private List<T> list;

        public OutList()
        {
            list = new List<T>();
        }

        public void add(T value)
        {
            list.Add(value);
        }

        public bool contains(T value)
        {
            return list.Contains(value);
        }

        public IEnumerable<T> getAll()
        {
            return list.AsEnumerable<T>();
        }

        public T getElement(int index)
        {
            return list[index];
        }

        public bool isEmpty()
        {
            return list.Count == 0;
        }

        public void remove(T value)
        {
            list.Remove(value);
        }

        public int size()
        {
            return list.Count();
        }

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            foreach (T el in list)
                sb.Append("" + el + ", ");
            return sb.ToString();
        }
    }
}

File: Map/Z_InterpretorWPro/src/domain/OutList.java
package domain;

import java.util.ArrayList;

public class OutList<T> implements IList<T>
{
	private ArrayList<T> al;
	
	public OutList()
	{
		al = new ArrayList<T>();
	}

	@Override
	public void add(T value) 
	{
		synchronized(this)
		{
			al.add(value);
		}
	}

	@Override
	public void remove(T value) 
	{
		synchronized(this)
		{
			al.remove(value);
		}
	}

	@Override
	public void remove(int index) 
	{
		synchronized(this)
		{
			al.remove(index);
		}
	}

	@Override
	public boolean isEmpty() 
	{
		synchronized(this)
		{
			return al.isEmpty();
		}
	}

	@Override
	public boolean contains(T value) 
	{
		synchronized(this)
		{
			return al.contains(value);
		}
	}

	@Override
	public T getElement(int index) 
	{
		synchronized(this)
		{
			return al.get(index);
		}
	}

	@Override
	public int size() 
	{
		synchronized(this)
		{
			return al.size();
		}
	}
	
	@Override
	public String toString()
	{
		synchronized(this)
		{
			StringBuffer sb = new StringBuffer();
			for(T e: al)
				sb.append(""+e+", ");
			return sb.toString();
		}
	}

	@Override
	public Iterable<T> getAll() 
	{
		synchronized(this)
		{
			return al;
		}
	}
	
	
}
